### PR TITLE
[Editor] Support for fault streams

### PR DESCRIPTION
**Pull request by lasanthaS**
## Purpose
This PR adds the design capabilities for Fault streams[1] in the Editor. 

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

## Related PRs
[1] https://github.com/siddhi-io/siddhi/pull/995/

### CODE REVIEW

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/StreamConfig.java [hunk 1]
```diff
@@ -73,6 +73,10 @@ public String getPartitionId() {
         return connectorsAndStreams;
     }
 
+    public boolean isFaultStream() {
+        return name.startsWith("!");
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/StreamConfig.java]: shall we add a constant for this

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js [hunk 2]
```diff
@@ -656,10 +656,23 @@ define(['require', 'log', 'jquery', 'backbone', 'lodash', 'designViewUtils', 'dr
...
+                            // Change the source id to the respective source id of the fault stream
+                            var streamMap = {};
+                            var eventStreamName = '';
+                            self.configurationData.getSiddhiAppConfig().getStreamList().forEach(function(s) {
+                                streamMap[s.getName()] = s;
+                                if (s.getId() === sourceId) {
+                                    eventStreamName = s.getName();
+                                }
+                            });
+                            sourceId = streamMap['!' + eventStreamName].getId();
```
Comment (dnwick) [inline components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js]: shall we use the defined constant for this

File: components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/StreamConfig.java (hunk 1 above)
Comment (lasanthaS) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/java/org/wso2/carbon/siddhi/editor/core/util/designview/beans/configs/siddhielements/StreamConfig.java]: Done

File: components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js (hunk 2 above)
Comment (lasanthaS) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js]: That constant has been added at the Java level. Will introduce a new one in JS level and use that here.

Comment (lasanthaS) [reply components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/design-view/design-grid.js]: Fixed in commit https://github.com/wso2/carbon-analytics/pull/1587/commits/7875d09ce7453c9f38bf7c70afe12336016d8908

**Pull request merged**